staging/src/k8s.io/client-go/examples/third-party-resources/apis/tpr/v1/types.go

```diff
@@ -23,25 +23,49 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
-type ExampleSpec struct {
-	Foo string `json:"foo"`
-	Bar bool   `json:"bar"`
-}
+const ExampleResourcePlural = "examples"
 
 type Example struct {
 	metav1.TypeMeta `json:",inline"`
-	Metadata        metav1.ObjectMeta `json:"metadata"`
+	// WARNING: Don't call the field below ObjectMeta to avoid the issue with ugorji
+	// https://github.com/kubernetes/client-go/issues/8#issuecomment-285333502
+	Metadata metav1.ObjectMeta `json:"metadata"`
```

Comment:
@sttts I think you meant keep it named ObjectMeta/ListMeta. Just checked, and yes, it seems to work now... Also it means there is no need of implementing `GetObjectKind` and other methods. Looks like some recent change has fixed this issue.
Will get rid of it then.